keras_cv/layers/preprocessing/grid_mask.py

```diff
@@ -141,43 +142,42 @@ def _compute_mask(self, image_height, image_width):
         if self.fill_mode == "constant":
             mask = tf.fill([mask_height, mask_width], value=-1)
         elif self.fill_mode == "gaussian_noise":
-            mask = tf.cast(tf.random.normal([mask_height, mask_width]), dtype=tf.int32)
+            mask = tf.cast(
+                self._random_generator.random_normal([mask_height, mask_width]),
+                dtype=tf.int32,
+            )
         else:
             raise ValueError(
                 "Unsupported fill_mode.  `fill_mode` should be 'constant' or "
                 "'gaussian_noise'."
             )
 
-        gridblock = tf.random.uniform(
-            shape=[],
-            minval=int(tf.math.minimum(image_height * 0.5, image_width * 0.3)),
-            maxval=int(tf.math.maximum(image_height * 0.5, image_width * 0.3)) + 1,
-            dtype=tf.int32,
-            seed=self.seed,
-        )
-
         if self.ratio == "random":
-            length = tf.random.uniform(
-                shape=[], minval=1, maxval=gridblock + 1, dtype=tf.int32, seed=self.seed
+            length = self._random_generator.random_uniform(
+                shape=[], minval=1, maxval=grid_block + 1, dtype=tf.int32
             )
         else:
             length = tf.cast(
                 tf.math.minimum(
                     tf.math.maximum(
-                        int(tf.cast(gridblock, tf.float32) * self.ratio + 0.5), 1
+                        int(tf.cast(grid_block, tf.float32) * self.ratio + 0.5), 1
                     ),
-                    gridblock - 1,
+                    grid_block - 1,
                 ),
                 tf.int32,
             )
 
         for _ in range(2):
             start_x = tf.random.uniform(
```

Comment:
should this use _random_generator too?

Reply:
Also - can we move compute mask to get_random_transformation or were there some sort of vectorization issues requiring it to be in the main func?